api/src/main/java/org/eclipse/microprofile/health/Response.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICES file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.microprofile.health;
+
+import org.eclipse.microprofile.health.spi.SPIFactory;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The response to a health check invocation.
+ * <p>
+ * The Response class is reserved for an extension by implementation providers.
+ * An application should use one of the static methods to create a Response instance using a ResponseBuilder.
+ * </p>
+ *
+ */
+public abstract class Response {
+
+    private static final Logger LOGGER = Logger.getLogger(Response.class.getName());
+
+    public static ResponseBuilder named(String name) {
+        ResponseBuilder builder = find(SPIFactory.class).createResponseBuilder();
+        return builder.name(name);
+    }
+
+    // the actual contract
+
+    public enum State { UP, DOWN }
+
+    public abstract String getName();
```

Comment:
I do not think that the name of the health check belongs to the `Response`.
I don't see an use case where a health check would return different names when it is called.

This may require a separate issue to discuss this but I think the `name` of the health check should be set on the `health check`. In the [https://docs.google.com/document/d/1p_5aNWX1A7d0waa0_JegYWm0iVxmtujoC-jNUoCNB7w/edit?disco=AAAABOB38Os](design document), I proposed to introduce a `@Health(name)` annotation to name a `HealthCheck` (and remove the info from the `Response`.

Please note that it's not clear to me what a name of a health check implies. 
As far as I can tell, nothing in the spec or the API mandates uniqueness of names.
But the [HTTP protocol document](https://github.com/eclipse/microprofile-evolution-process/blob/master/proposals/0003-spec.md#protocol-adaptor) implies otherwise:
> The JSON response MUST contain the `id` entry specifying the name of the check, to support protocols that support external identifier (i.e. URI)